src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -570,6 +571,35 @@ def execute_tool(self, function_name, arguments):
         """
         logging.debug(f"{self.name} executing tool {function_name} with arguments: {arguments}")
 
+        # Check if approval is required for this tool
+        from ..approval import is_approval_required, console_approval_callback, get_risk_level, mark_approved, ApprovalDecision
```

Ruff (0.11.9)

`..approval.ApprovalDecision` imported but unused

Remove unused import: `..approval.ApprovalDecision`

(F401)